scene/gui/color_picker.cpp

```diff
@@ -194,72 +198,35 @@ void ColorPicker::set_focus_on_line_edit() {
 }
 
 void ColorPicker::_update_controls() {
-	const char *rgb[3] = { "R", "G", "B" };
-	const char *hsv[3] = { "H", "S", "V" };
-	const char *hsl[3] = { "H", "S", "L" };
-	if (hsv_mode_enabled && picker_type == SHAPE_OKHSL_CIRCLE) {
-		for (int i = 0; i < 3; i++) {
-			labels[i]->set_text(hsl[i]);
-		}
-	} else if (hsv_mode_enabled && picker_type != SHAPE_OKHSL_CIRCLE) {
-		for (int i = 0; i < 3; i++) {
-			labels[i]->set_text(hsv[i]);
-		}
-	} else {
-		for (int i = 0; i < 3; i++) {
-			labels[i]->set_text(rgb[i]);
-		}
+	int mode_sliders_count = selected_mode->get_slider_count();
+
+	for (int i = current_slider_count; i < mode_sliders_count; i++) {
+		sliders[i]->show();
+		labels[i]->show();
+		values[i]->show();
 	}
-	if (picker_type == SHAPE_OKHSL_CIRCLE) {
-		btn_hsv->set_text(RTR("OKHSL"));
-	} else {
-		btn_hsv->set_text(RTR("HSV"));
-	}
-	if (hsv_mode_enabled) {
-		set_raw_mode(false);
-		set_hsv_mode(true);
-		btn_raw->set_disabled(true);
-	} else if (raw_mode_enabled) {
-		set_raw_mode(true);
-		set_hsv_mode(false);
-		btn_raw->set_disabled(false);
-		btn_hsv->set_disabled(true);
-	} else {
-		set_raw_mode(false);
-		set_hsv_mode(false);
-		btn_raw->set_disabled(false);
-		btn_hsv->set_disabled(false);
-	}
-
-	if (raw_mode_enabled) {
-		for (int i = 0; i < 3; i++) {
-			scroll[i]->remove_theme_icon_override("grabber");
-			scroll[i]->remove_theme_icon_override("grabber_highlight");
-			scroll[i]->remove_theme_style_override("slider");
-			scroll[i]->remove_theme_style_override("grabber_area");
-			scroll[i]->remove_theme_style_override("grabber_area_highlight");
-		}
-	} else {
-		Ref<StyleBoxEmpty> style_box_empty(memnew(StyleBoxEmpty));
-		Ref<Texture2D> bar_arrow = get_theme_icon(SNAME("bar_arrow"));
-
-		for (int i = 0; i < 4; i++) {
-			scroll[i]->add_theme_icon_override("grabber", bar_arrow);
-			scroll[i]->add_theme_icon_override("grabber_highlight", bar_arrow);
-			scroll[i]->add_theme_style_override("slider", style_box_empty);
-			scroll[i]->add_theme_style_override("grabber_area", style_box_empty);
-			scroll[i]->add_theme_style_override("grabber_area_highlight", style_box_empty);
-		}
+	for (int i = mode_sliders_count; i < current_slider_count; i++) {
+		sliders[i]->hide();
+		labels[i]->hide();
+		values[i]->hide();
+	}
+	current_slider_count = mode_sliders_count;
+
+	for (int i = 0; i < current_slider_count; i++) {
+		labels[i]->set_text(selected_mode->get_slider_label(i));
 	}
+	alpha_label->set_text(RTR("A"));
```

Comment:
Does translating it even make sense? 🤔
And what about other labels? You obviously can't RTR them (because the translation will not be fetched automatically).

Reply:
Okay then, removing the RTR call.